enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+    subgraph MachineOSConfig [MachineOSConfig]
+        Secrets("Push and pull secrets")
+        FinalImagePullspec("Final image pullspec")
+        Containerfile("User-provided Containerfile")
+        MCPName("MachineConfigPool Name")
+    end
+    
+    subgraph MachineConfigPool [MachineConfigPool]
+        RenderedMachineConfig("Rendered MachineConfig")
+    end
+    
+    subgraph MachineOSBuild [MachineOSBuild]
+        MCPName("MachineConfigPool Name")
+        CurrentConfig("Current MachineConfig for pool")
+        Status("Current status of build")
+    end
+       
+    MachineOSConfig-->|A MachineOSConfig or change thereof produces a MachineOSBuild|MachineOSBuild
+    MachineConfigPool-->|A MachineConfig change produces a MachineOSBuild|MachineOSBuild
+    MachineOSBuild-->|The MachineOSBuild starts a build|BuildProcess
+    BuildProcess("Build Process")-->|After completion, the NodeController is notified|NodeController
+    NodeController-->|NodeController rolls out the newly-built image to all of the nodes|Nodes
+```
+
+#### MachineOSConfig
+
+This is where the knobs that control the Machine OS Build process would live. Most of these would be provided during the one-time setup phase:
+
+| **Field Name / Path (JSONPath):**                 | **Optional or Required:** | **Intended Use:**                                                                                                                                                                                                                                                             |
+| ------------------------------------------------- | ------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.spec.machineConfigPool.name`                    | Required                  | The name of the MachineConfigPool whose rendered MachineConfig will be built into the image.                                                                                                                                                                                  |
+| `.spec.buildInputs.baseOSExtensionsImagePullspec` | Optional                  | The pullspec of the extensions image. If not provided, this value will come from the system defaults found in the <add me> ConfigMap                                                                                                                                          |
+| `.spec.buildInputs.baseOSImagePullspec`           | Optional                  | The pullspec of the base OS image. If not provided, this value will come from the system defaults found in the <add me> ConfigMap                                                                                                                                             |
```

Comment:
Why would defaults come from a configmap? They are better coming from a CR of some description, however, I'm not sure defaulting them from another object is desirable at all, what if there is a race condition?

Why are system defaults ever needed here? Why not just require the fields?